juju_verify/cli.py

```diff
@@ -80,10 +80,33 @@ def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, self.dest, items)
 
 
+def parse_charm_mapping(parsed_args: argparse.Namespace) -> None:
```

Comment:
I do not like this approach, where input is actually output from `parser.parse_args()`. IMO we should integrated it to parsing process, with creating custom `Action` or use `type=<validation_function>`. e.g.
```python
def charm_mapping(mapping: str) -> Tuple[str, BaseVerifier]:
    """Charm mapping type."""
    if not isinstance(mapping, str):
        raise ValueError("Mapping object is not string")

    try:
        charm, verifier = mapping.split(":")
        verifier = SUPPORTED_CHARMS[verifier]
        return charm, verifier
    except ValueError as error:
        raise ValueError(f"Unsupported `{mapping}` mapping format. "
                         "expected <APP_NAME>:<CHARM_NAME>") from error
    except KeyError as error:
        raise ValueError(f"Unsupported charm. {SUPPORTED_CHARMS.keys()}") from error
```

Reply:
Cool feature of `argparse`, did not know about this. I like the `type=` approach with custom parser

Reply:
Cool feature of `argparse`, did not know about this. I like the `type=` approach with custom parser

Reply:
One note: I would not put the verification of Supported/Unsupported charm here. We already do that in `get_verifier` function and we can keep doing it there for both implicitly and explicitly defined charms.
The `get_verifier` is bit further down the execution flow but we wont have to maintain two error outputs.

Reply:
> One note: I would not put the verification of Supported/Unsupported charm here. We already do that in get_verifier function and we can keep doing it there for both implicitly and explicitly defined charms.
The get_verifier is bit further down the execution flow but we wont have to maintain two error outputs.

There is only an [error message](https://github.com/canonical/juju-verify/blob/17fb19db59380245f5bdee80020571e03f9deb42/juju_verify/verifiers/__init__.py#L68), but I understand your point here and I agree.